File: leetcode/225.1.cpp
class MyStack {
private:
	queue<int> d1;
	queue<int> d2;
	queue<int> *p1;
	queue<int> *p2;

public:
	/** Initialize your data structure here. */
	MyStack() {
		p1 = &d1;
		p2 = &d2;
	}

	/** Push element x onto stack. */
	void push(int x) {
		p2->push(x);
		while (!p1->empty()) {
			p2->push(p1->front());
			p1->pop();
		}

		queue<int> *tmp = p1;
		p1 = p2;
		p2 = tmp;
	}

	/** Removes the element on top of the stack and returns that element. */
	int pop() {
		int ret = p1->front();
		p1->pop();
		return ret;
	}

	/** Get the top element. */
	int top() {
		return p1->front();
	}

	/** Returns whether the stack is empty. */
	bool empty() {
		return p1->empty();
	}
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */

File: leetcode/225.1.java
class MyStack {
    private Queue<Integer> data1;
    private Queue<Integer> data2;

    /**
     * Initialize your data structure here.
     */
    public MyStack() {
        data1 = new LinkedList<Integer>();
        data2 = new LinkedList<Integer>();
    }

    /**
     * Push element x onto stack.
     */
    public void push(int x) {
        data2.add(x);
//         for(int i = 0; i<data1.size(); i++){
//             data2.add(data1.poll());
//         }
        while (data1.size() > 0) {
            data2.add(data1.poll());
        }
        Queue<Integer> tmp = data1;
        data1 = data2;
        data2 = tmp;
    }

    /**
     * Removes the element on top of the stack and returns that element.
     */
    public int pop() {
        return data1.poll();
    }

    /**
     * Get the top element.
     */
    public int top() {
        return data1.peek();
    }

    /**
     * Returns whether the stack is empty.
     */
    public boolean empty() {
        return data1.size() == 0;
    }
}